### DISCUSSION TITLE
Graphs not being plotted

### DISCUSSION THREAD

**Discussion by bengladwyn** [Part-II-Computational-Physics/.github#4] [General]
I did the pip install for ipyml in the terminal, and restarted then ran the code.

I get "'%matplotlib' widget works best inside of VS Code" but it makes no difference as it defaults to notebook. Using %matplotlib ipyml gives an error. 

No graph is plotted, it just says: "<IPython.core.display.HTML object>"

How do I get this to work?

**Comment by AustenLamacraft** [comment]
Possibly you have more than one Python distribution installed. If you are using Anaconda you could try 

`conda install -c conda-forge ipympl`

**Comment by DaanTimmers** [comment]
I think I'm having a similar issue. I'm using Anaconda and used the command posted above. I have no reason to believe the install wasn't successful, but when I run <code> %matplotlib ipympl </code> I get 'Warning: Cannot change to a different GUI toolkit: ipympl. Using notebook instead.' as output. Using notebook however, doesn't output any figures. If I then use <code> %matplotlib inline </code> as the VS code editor recommends, figures are produced but none of the animations work (they just show the figure at t/T = 0.00).

**Reply by AustenLamacraft** [reply]
Do you get that message in a blank notebook with just one cell containing `%matplotlib ipympl`?

**Reply by bengladwyn** [reply]
Yep, same here. Managed to get images appearing inline when ipympl is used, but can't change to notebook and no animations show. It worked briefly earlier but closed VS code and its broken again.

**Comment by Takodachi6969** [comment]
I solved the problem by deleting the first line in the code
'# %matplotlib notebook '
to plot a graph. 

The animation could be called by using 
from matplotlib import rc
rc('animation', html='jshtml')
and calling the anim at the end, but it takes forever to do. (Same as when trying to savefig)

The code provided couldn't render the animation.

**Reply by cl804** [reply]
Similar problem, for the animation the number is changing but the graph is not

**Reply by MajaR01** [reply]
I have the opposite - the number is not changing, but the graph is

**Comment by cl804** [comment]
The way I solved the problem is to change to Jupyter Notebook. Looks like Vscode doesn't work well for animations.

**Comment by Takodachi6969** [comment]
The problem was solved by increasing the number of steps, rendering fewer frames, and increasing the interval to make it readable.

```
 num_frames = 100
 def animate(i):
     for _ in range(num_frames):
       step(f)

     show_ft.set_data(X, f)
    
     time.set_text("t / T = %.2f" % (v * i * dt / (x_max - x_min)))

     return show_f0, show_ft
 anim = animation.FuncAnimation(fig, animate, 10**5//num_frames, interval = dt * 800, blit = True)
 HTML(anim.to_html5_video())

```

It took about 60 seconds to render, with no memory overflow

**Comment by MajaR01** [comment]
I cannot run a single cell in VS code, after I connected it to GitHub Classroom - it opens a prompt to pick a Kernel, but when I tried it just ignores it and nothing happens

**Comment by sebastianl2014** [comment]
Probably something suboptimal with this approach but tried adding display(anim) to the end of the code and that seemed to provide an animation. Although would appreciate an explanation of how I could make plt.show actually show the animation as intended?

**Comment by NathanMagnan** [comment]
Ok, so its seems that the animation is not working for a lot of people, that's annoying 
It's probably due to the many different distributions of Python that one could have.

My advice would be to:
- Try to do the exercise on jupyter notebook instead of VS code, because I tested the notebook only within jupyter
- If you really want to use VS code, try deleting the `%matplotlib notebook` command in the first cell, beccause it is a magic command of jupyter that is gonna break in VS code

I hope this helps?

**Comment by AustenLamacraft** [comment]
Don't worry if you can't get the animation to work: to answer the questions you can step through the solver and then plot: by evolving through different times (different numbers of periods) you'll be able to see how the solution behaves
